federation/planner.go

```diff
@@ -56,8 +57,30 @@ type Plan struct {
 // This breaks every query into subqueries that can each be resolved by a single graphQLServer
 // and describes what sub-queries need to be resolved first.
 type Planner struct {
-	schema    *SchemaWithFederationInfo //schema describes what fields the graphql servers know about along with the services that know how to execute each field
-	flattener *flattener                //flattener knows how to combine all the fragments on a query into a singel query
+	// schema describes what fields the graphql servers know about along with the services that know how to execute each field.
+	schema *SchemaWithFederationInfo
+	// flattener knows how to combine all the fragments on a query into a singel query.
+	flattener       *flattener
+	serviceSelector ServiceSelector
+}
+
+// serviceSelector is an optional field which can be used to override the <type,field> to service mapping,
+// which is useful during the fieldfunc migration.
+type ServiceSelector func(typeName string, fieldName string) string
```

Comment:
Can we allow the service selector to return a descriptive error instead of an empty string when things go wrong

Reply:
I respond this in the comment below.